test/Base/MemoryZip.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/python2
+
+# Copyright (c) 2017 Public Library of Science
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+__author__ = 'rskonnord'
+
+"""Utility for creating zip files in memory."""
+
+import io
+import zipfile
+
+class MemoryZipEntry(object):
+    """An entry to be inserted into an in-memory zip file."""
+
+    def __init__(self, name):
+        self.name = name
+
+    def get_file(self):
+        return io.BytesIO(self._get_file_bytes())
+
+class MemoryZipData(MemoryZipEntry):
+    """An entry containing data that is already in memory."""
+
+    def __init__(self, name, data):
+        MemoryZipEntry.__init__(self, name)
+        self._data = data
+
+    def _get_file_bytes(self):
+        if isinstance(self._data, str):
```

Comment:
docstring. Also tell why this is a internal method (because the _) and why it is being used from another class